### ISSUE TITLE
Is using Doctrine ODM possible ?

### ISSUE THREAD

**Issue by etshy** [thephpleague/oauth2-server-bundle#103]
I only saw ORM references to store token but I prefer ask in case I missed something.
There is currently no way to use Doctrine ODM right ?

**Comment by imdhemy** [comment]
Hey @etshy

Did you come across an answer to your question anywhere else? If so, please let us know 🙌

**Comment by chalasr** [comment]
Indeed. I won't work on this myself, but I would welcome a PR adding support for the ODM.

**Comment by etshy** [comment]
I won't garantee I'll do it but I'll take a look. (I started to make my own implementation of oauth2-server for my current project though)

**Comment by imdhemy** [comment]
I accidentally delete my previous comment.

---
@etshy 

What do you think about using:

**For Tokens**
- [LexikJWTAuthenticationBundle](https://symfony.com/bundles/LexikJWTAuthenticationBundle/current/index.html) 
- [JWTRefreshTokenBundle](https://github.com/markitosgv/JWTRefreshTokenBundle)

**For clients**
- Implement them in an ordinary way
- Set up [filters](https://symfony.com/doc/current/event_dispatcher/before_after_filters.html#how-to-set-up-before-and-after-filters) to validate and authenticate clients.

**Comment by etshy** [comment]
@imdhemy You mean for my implementation ?
I already choose to use Lexik bundle to create the token, and add data to it.
I didn't know about the RefreshTokenBundle though, thanks a lot.

**Comment by etshy** [comment]
Hi. No sorry I put the project I planned to make a oauthserver for on hold so I didn't do anything about it yet.

**Comment by etshy** [comment]
I quickly took a look at this tonight and as a complete newbie with Bundle, I'm a bit lost.
Should I add both ORM and ODM compilerPass or should I move that to the `prepend` to be able to load only one base on the Configuration ? I made some few tests but adding `CompilerPass` in the `prepend` doesn't seem to be working at all though.

So I guess I should load both and in the `configureDoctrinePersistence` I replace argument by `entityManager` or `documentManager` depending on the Configuration.

Another thing, Should I rework the Driver to replace it by files Mapping (XML ?) ? I saw in the `DoctrineOrmMappingsPass` that "You should not directly instantiate this class but use one of the factory methods."
